### DISCUSSION TITLE
Does Blosc2 support asynchronous codec

### DISCUSSION THREAD

**Discussion by boxerab** [Blosc/c-blosc2#620] [General]
When encoding on an accelerator, not CPU, the most efficient way is often to schedule the work and to let the accelerator trigger an event when encoding is done. Does Blosc2 support this work flow ? i.e. schedule a block on a codec plugin and get notification when encoding is complete.

**Comment by FrancescAlted** [comment]
It does not, I'm afraid.

**Comment by boxerab** [comment]
Thanks. What would be involved in supporting asynch ?  I imagine that a lot of code expects the plugin to return immediately with the encoded data.
